install.sh

```
@@ -108,104 +91,136 @@ download_file() {
+        print_color $YELLOW "Removing $file (requires sudo)..."
+        sudo rm -f "$file"
+    fi
 }
 
 # Function to install system-wide
 install_system() {
-    check_permissions "system"
+    print_color $BLUE "Installing git-commitai..."
+
+    # Create temp directory for downloads
+    local temp_dir=$(mktemp -d)
```

Shellcheck (0.10.0)

[warning] Declare and assign separately to avoid masking return values.

(SC2155)